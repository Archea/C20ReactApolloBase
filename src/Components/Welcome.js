import React, { Component } from 'react'
import { Query } from 'react-apollo'
import gql from 'graphql-tag'

export const GET_USERDATA = gql`
  {
    userdata @client {
      name
      firm {
        id
        name
      }
      master {
        id
        name
      }
    }
  }
`

export class Welcome extends Component {
  render() {
    return (
      <Query query={GET_USERDATA}>
        {({ loading, error, data }) => {
          if (error) return `Error! ${error.message}`
          if (loading) return 'loading...'

          return (
            <div className="Welcome">
              <h2>Welcome {data.userdata.name}!</h2>
            </div>
          )
        }}
      </Query>
    )
  }
}

export default Welcome
