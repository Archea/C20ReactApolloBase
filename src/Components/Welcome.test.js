import React from 'react'
import ReactDOM from 'react-dom'
import { Welcome, GET_USERDATA } from './Welcome'
import renderer from 'react-test-renderer'
import { MockedProvider } from 'react-apollo/test-utils'

const mocks = [
  {
    request: {
      query: GET_USERDATA,
      variables: {}
    },
    result: {
      data: {
        userdata: {
          name: 'Anonymous User',
          firm: {
            id: 'ec7782f7-d919-4a58-bb8e-fe888633481f',
            name: 'Avitru'
          },
          master: {
            id: 'f069a64d-01d0-449b-89c4-6fbe3571af64',
            name: 'MasterSpec'
          }
        }
      }
    }
  }
]

describe('Welcome', () => {
  const mockprops = {
    client: {}
  }
  test('it at renders and matches snapshot', () => {
    const comp = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Welcome {...mockprops} />
      </MockedProvider>
    )
    expect(comp).toBeDefined()
    expect(comp.toJSON()).toMatchSnapshot()
  })
})
