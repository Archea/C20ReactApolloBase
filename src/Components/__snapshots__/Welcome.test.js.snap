// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome it at renders and matches snapshot 1`] = `"loading..."`;
